internal/cmd/config/config_test.go

```diff
@@ -524,3 +524,58 @@ func TestController_EventingConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestParsingPath(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name      string
+		in        string
+		expOut    string
+		expErrStr string
+	}{
+		{
+			name:      "env: not present",
+			in:        "env://TEST_ENV_FIELD_THAT_SURELY_DOESNT_EXIST",
+			expOut:    "",
+			expErrStr: "",
+		},
+		{
+			name:      "file: not present",
+			in:        "file://test_file_that_surely_doesnt_exist",
+			expOut:    "file://test_file_that_surely_doesnt_exist",
+			expErrStr: "error reading file at file://test_file_that_surely_doesnt_exist: open test_file_that_surely_doesnt_exist: no such file or directory",
+		},
+		{
+			name:      "not a url",
+			in:        "some-value",
+			expOut:    "some-value",
+			expErrStr: "",
+		},
+		{
+			name:      "upper case value",
+			in:        "some-VALUE",
+			expOut:    "some-VALUE",
+			expErrStr: "field must be all lower-case",
+		},
+		{
+			name:      "non-printable value",
+			in:        "some-val\u0000ue",
+			expOut:    "some-val\u0000ue",
+			expErrStr: "field contains non-printable characters",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			out, err := parsePath(tt.in)
+			if len(tt.expErrStr) > 0 {
```

Comment:
nit: in most of the test I have encountered we create a `bool` to explicitly set the expectation of an error instead of checking the string.  I feel this helps in ready the test, usually its `wantErr` but `expErr` should be good

Reply:
In those tests, do we assert the error itself? 
i.e: Do we have a boolean to assert whether we expect an error or not and _then_ we check against some other field to assert the correct_error was thrown, or do we just use the boolean to say _an_ error _should_ be thrown?

Reply:
Great question, so we generally have the following
1) Use a bool to set expectation of an error or not. In the test we then ‘if-else’ that book and require error or noError
2) Inside the ‘if’ we often also have a error ‘Is’ check.  In this case you won’t be able to check that
3) Finally we often have an error string check like you did here with ErrorEqual. Sometimes we do not do this as it can make the tests brittle (if the error string changes all tests have to change, and any test in files consuming this package higher up might have to change).  However, I prefer to add the error string check to tests because it helps me visualize the full error message.  In this case since the errors are in the cmd package I would argue we should definitely assert the error string

Reply:
Makes sense, thanks for clarifying! 🙇 

With that in mind, we should add a `wantErr` or `expErr` boolean as that'll probably improve readability later down in the test exec and then when we _do_expect an error, match against `wantErrStr` or `expErrStr` to ensure it's the expected one.

WDYT?

Reply:
SGTM. I have a slight preference to want, as I feel it reads better but exp is used in other tests in this package so as you prefer